### PR TITLE
test(functions/firebase)

### PR DESCRIPTION
**Pull request by rluvaton**
Add firebase tests and input validation in the firebase service

### CODE REVIEW

File: functions/services/firebase/firebase.test.js [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+        });
+
+        afterEach(() => {
+            writeNewTeam = undefined;
+        });
+
+        it("should write new team when Firestore is empty", async () => {
+            let firestoreTeamsData;
+
+            let teamData = { OnFleetID: "<some-fleet-id>", zipcode: "342897" };
```
Comment (edwinthinks) [inline functions/services/firebase/firebase.test.js]: I've recently added the faker library. So you could randomize the values like this:
```suggestion
            let teamData = { OnFleetID: faker.random.number(), zipcode: faker.address.zipCode() };
```

However, you must add `const faker = require('faker')` to the top.

There is a few places that you could utilize this :) and I'll leave it to you to utilize them where you see fit in the PR.

File: functions/services/firebase/firebase.test.js [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+            expect(pr).toResolve();
+
+            // running the waiting tasks (without flush the pr will never resolve)
+            firestore.flush();
+            const team = await pr;
+
+            expect(team).toBeNull();
+        });
+
+        it("should not get team when Firestore already different team", async () => {
```
Comment (edwinthinks) [inline functions/services/firebase/firebase.test.js]:
```suggestion
        it("should not get team when Firestore already has different team", async () => {
```
? Not sure if this was the intention

File: functions/services/firebase/firebase.test.js (hunk 1 above)
Comment (rluvaton) [reply functions/services/firebase/firebase.test.js]: Thank you, I will

File: functions/services/firebase/firebase.test.js (hunk 2 above)
Comment (rluvaton) [reply functions/services/firebase/firebase.test.js]: Yeah, thank you

**Pull request merged**